homeassistant/components/wemo/wemo_device.py

```diff
@@ -24,23 +24,23 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-class DeviceWrapper:
+class DeviceCoordinator(DataUpdateCoordinator):
     """Home Assistant wrapper for a pyWeMo device."""
 
-    def __init__(
-        self,
-        hass: HomeAssistant,
-        wemo: WeMoDevice,
-        device_id: str,
-        coordinator: DataUpdateCoordinator,
-    ) -> None:
-        """Initialize DeviceWrapper."""
+    def __init__(self, hass: HomeAssistant, wemo: WeMoDevice, device_id: str) -> None:
+        """Initialize DeviceCoordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=wemo.name,
+            update_interval=timedelta(seconds=30),
+            update_method=self.async_update_data,
```

Comment:
We can rename `async_update_data` to `_async_update_data` and drop the parameter here. That method is the coordinator internal update method which is overridable.

Reply:
Done.